src/app/services/downloads.service.ts

```diff
@@ -0,0 +1,64 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, timer } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ExportResult, QhanaBackendService } from './qhana-backend.service';
+
+interface ExportCache {
+    exportId: number;
+    status: "SUCCESS" | "FAILURE" | "PENDING";
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class DownloadsService {
+    private exportsCache: ExportCache[] | null = null;
+    private exportList: BehaviorSubject<ExportResult[] | null> = new BehaviorSubject<ExportResult[] | null>(null);
+    private downloadsCounter: BehaviorSubject<number> = new BehaviorSubject(0);
+
+    constructor(private backend: QhanaBackendService) {
+        timer(0, 2000).subscribe(() => {
```

Comment:
every 2 seconds is a bit frequent for stuff like this. Every 5 Minutes standard and more frequent when things are ongoing would be preferable (but also more complex). Minimum is to reduce the polling frequency!